### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by erikd**

### CODE REVIEW

File: ouroboros-consensus/src/Ouroboros/Consensus/Protocol/Praos.hs [hunk 1]
```diff
@@ -136,8 +136,11 @@ forgePraosFields PraosNodeConfig{..} PraosProof{..} mkToSign = do
            (mkToSign signedFields)
            keyKES
     case m of
-      Nothing                  -> error "mkOutoborosPayload: signedKES failed"
-      Just (signature, newKey) -> do
+      Nothing -> error "mkOutoborosPayload: signedKES failed"
+      Just signature -> do
+        -- TODO : We should not update the key on each signing, but X slots
+        -- (for configurable param X)
+        newKey <- fromMaybe (error "mkOutoborosPayload: updateKES failed") <$> updateKES () keyKES
```
Comment (dcoutts) [inline ouroboros-consensus/src/Ouroboros/Consensus/Protocol/Praos.hs]: @nc6 can you have a look at this. The KES API has been fixed and so this bit here needs updating. We need: 1. to separate key evolution from block signing, we only evolve the key based on slots (slots equivalent to 24 hrs in Shelley), and 2. KES evolutions are finite, when we run out we can not longer sign blocks, and we need a new operational cert.

Comment (edsko) [reply ouroboros-consensus/src/Ouroboros/Consensus/Protocol/Praos.hs]: Note that this is the mock Praos implementation, not the real one.

Comment (erikd) [reply ouroboros-consensus/src/Ouroboros/Consensus/Protocol/Praos.hs]: :man_shrugging: I have no idea what I am doing this far down the stack.

Comment (edsko) [reply ouroboros-consensus/src/Ouroboros/Consensus/Protocol/Praos.hs]: No, of course, I'd not expect anyone to be able to apply a real update in the underlying crypto interface to make changes to a specific (and non-trivial) consensus protocol :)

**Comment by edsko** [comment]
Created a ticket for re-enabling these tests at https://github.com/input-output-hk/ouroboros-network/issues/1545 .

**Pull request merged**